core/arch/arm/plat-rockchip/platform_config.h

```diff
@@ -1,94 +1,112 @@
-
-#define GICC_BASE		(GIC_BASE + GICC_OFFSET)
-#define GICD_BASE		(GIC_BASE + GICD_OFFSET)
+#define GIC_SIZE		SIZE_K(64)
+#define GICD_BASE		(GIC_BASE + 0x1000)
+#define GICC_BASE		(GIC_BASE + 0x2000)
 
 #define SGRF_BASE		0x10140000
+#define SGRF_SIZE		SIZE_K(64)
+
 #define DDRSGRF_BASE		0x10150000
+#define DDRSGRF_SIZE		SIZE_K(64)
+
 #define GRF_BASE		0x11000000
+#define GRF_SIZE		SIZE_K(64)
+
 #define UART2_BASE		0x11030000
-#define CRU_BASE		0x110E0000
+#define UART2_SIZE		SIZE_K(64)
+
+#define CRU_BASE		0x110e0000
+#define CRU_SIZE		SIZE_K(64)
 
 /* Internal SRAM */
 #define ISRAM_BASE		0x10080000
-#define ISRAM_SIZE		0x8000
+#define ISRAM_SIZE		SIZE_K(8)
+
+#elif defined(PLATFORM_FLAVOR_rk3399)
+
+#define MMIO_BASE		0xF8000000
+
+#define GIC_BASE		(MMIO_BASE + 0x06E00000)
+#define GIC_SIZE		SIZE_M(2)
+#define GICD_BASE		(GIC_BASE)
+#define GICR_BASE		(GIC_BASE + SIZE_M(1))
+
+#define UART0_BASE		(MMIO_BASE + 0x07180000)
+#define UART0_SIZE		SIZE_K(64)
+
+#define UART1_BASE		(MMIO_BASE + 0x07190000)
+#define UART1_SIZE		SIZE_K(64)
+
+#define UART2_BASE		(MMIO_BASE + 0x071A0000)
+#define UART2_SIZE		SIZE_K(64)
+
+#define UART3_BASE		(MMIO_BASE + 0x071B0000)
+#define UART3_SIZE		SIZE_K(64)
 
-/* Periph IO */
-#define PERIPH_BASE		0x10100000
-#define PERIPH_SIZE		0x22000000
+#define SGRF_BASE		(MMIO_BASE + 0x07330000)
+#define SGRF_SIZE		SIZE_K(64)
+
+#elif defined(PLATFORM_FLAVOR_px30)
+
+#define GIC_BASE		0xff130000
+#define GIC_SIZE		SIZE_K(64)
+#define GICD_BASE		(GIC_BASE + 0x1000)
+#define GICC_BASE		(GIC_BASE + 0x2000)
+
+#define UART1_BASE		0xff158000
+#define UART1_SIZE		SIZE_K(64)
+
+#define UART2_BASE		0xff160000
+#define UART2_SIZE		SIZE_K(64)
+
+#define UART5_BASE		0xff178000
+#define UART5_SIZE		SIZE_K(64)
+
+#define FIREWALL_DDR_BASE	0xff534000
+#define FIREWALL_DDR_SIZE	SIZE_K(16)
 
 #else
 #error "Unknown platform flavor"
 #endif
 
-#define CONSOLE_UART_BASE	UART2_BASE
-#define CONSOLE_BAUDRATE	1500000
-#define CONSOLE_UART_CLK_IN_HZ	24000000
-
+#if defined(PLATFORM_FLAVOR_rk322x)
+/* Keep hard-coded old behaviour for rk322x */
+#define EARLY_CONSOLE_UART_BASE		UART2_BASE
+#define EARLY_CONSOLE_UART_SIZE		UART2_SIZE
+#define EARLY_CONSOLE_BAUDRATE		1500000
+#define EARLY_CONSOLE_UART_CLK_IN_HZ	24000000
+#else
 /*
- * Rockchip memory map
- *
- * +---------------------------+
- * |        | TEE_RAM |  1 MiB |
- * + TZDRAM +------------------+
- * |        | TA_RAM  |  1 MiB |
- * +--------+---------+--------+
- * | SHMEM  |         |  1 MiB |
- * +---------------------------+
+ * Early Console UART definitions
+ * OP-TEE will switch to a devicetree-based console later during
+ * startup, based on the actual board-specific console settings.
+ * So the fixed early console should only be enabled for debugging
+ * purposes during development:
+#define EARLY_CONSOLE_UART_BASE		UART2_BASE
+#define EARLY_CONSOLE_UART_SIZE		UART2_SIZE
+#define EARLY_CONSOLE_UART_CLK_IN_HZ	24000000
+#define EARLY_CONSOLE_BAUDRATE		1500000
```

Comment:
Strange way for a comment. Could be confusing. @jforissier, would such comment be fine?

Reply:
Agreed, if the purpose is to be able to enable or disable early console at compile time, while keeping the default behavior for `rk322x`, then a `CFG_` flag would be more suitable:
- In `plat-rockchip/conf.mk`, set `CFG_EARLY_CONSOLE ?= y` in the `rk322x` section (optional, that's if you want to keep the current behavior which I think is a reasonable thing to do). Add `CFG_EARLY_CONSOLE ?= n` immediately after, and move the comment about device tree and debugging at that location.
- Use `#ifdef CFG_EARLY_CONSOLE` instead of `#ifdef EARLY_CONSOLE_UART_BASE` in the `.c` files.
- Remove the commented out section here, and adjust the `#if defined(PLATFORM_FLAVOR_rk322x)` to fit all platform flavors for which those UART values are valid (you can just remove it if all flavors can use the same values).

Reply:
The core issue is that right now we have support for all boards per soc in the flavor (so all boards using rk322x, all using rk3399, etc). As the comment indicates the debug uart can vary per board.

It relies on how the debug uart was setup during the previous loader stage, so using the wrong uart can actually result in hangs. Therefore the early uart should be off in general, except people knowing what they're doing.

But you're right, these values can actually move to conf.mk as CFG_EARLY_UART_BASE etc, so no need for meddling with the plaform_config.h